lib/core/Core.ts

```diff
@@ -32,16 +33,18 @@ export default class Core {
   /**
    * Core constructor.
    */
-  public constructor (config: Config, versionModels: VersionModel[], private cas: ICas) {
+  public constructor (private config: Config, versionModels: VersionModel[], private cas: ICas) {
     // Component dependency construction & injection.
-    this.versionManager = new VersionManager(config, versionModels); // `VersionManager` is first constructed component.
+    this.versionManager = new VersionManager(config, versionModels); // `VersionManager` is first constructed component as multiple components depend on it.
+    this.serviceInfo = new ServiceInfo('core');
     this.operationStore = new MongoDbOperationStore(config.mongoDbConnectionString, config.databaseName);
     this.blockchain = new Blockchain(config.blockchainServiceUri);
     this.downloadManager = new DownloadManager(config.maxConcurrentDownloads, this.cas);
     this.resolver = new Resolver(this.versionManager, this.operationStore);
-    this.batchScheduler = new BatchScheduler(this.versionManager, this.blockchain, config.batchingIntervalInSeconds);
     this.transactionStore = new MongoDbTransactionStore(config.mongoDbConnectionString, config.databaseName);
     this.unresolvableTransactionStore = new MongoDbUnresolvableTransactionStore(config.mongoDbConnectionString, config.databaseName);
+
+    this.batchScheduler = new BatchScheduler(this.versionManager, this.blockchain, config.batchingIntervalInSeconds);
     this.observer = new Observer(
```

Comment:
Should we conditionally create the batch scheduler and observer so we don't have to have a transaction store or blockchain service ready for it if we don't want to in the future?

Reply:
Good suggestion, I prefer to kick this can down the road because the change right now is super simple and in one place. If we now say we will not create observer if observing is disabled, then it begs the follow up question such as "do we need transaction store?" probably not, do we then remove the instantiation of the stores too? Now `VersionManager` references it, do we then do plumb through the `VersionManager`? Just too much than what is worth right now IMO.

Also with the current approach, we leave the possibility open for the observer/writer to start polling through an API call, might be useful for live site mitigation.